### PR TITLE
Address LineStringSegmentize bugs

### PR DESCRIPTION
**Pull request by JosiahParry**
- [x] I agree to follow the project's [code of conduct](https://github.com/georust/geo/blob/main/CODE_OF_CONDUCT.md).
- [x] I added an entry to `CHANGES.md` if knowledge of this change could be valuable to users.
---

Thanks to @Robinlovelace for doing tests on _real_ data. Brought to light 2 categories of edge cases where the number of LineStrings in the resultant MultiLineString was not the same quantity as requested by the `n` parameter.

Bug occurred when: 

- There were only 2 coordinates in the LineString and
- When the `Line` length was larger than the maximum size of a segment

This has been addressed through the use of the `Densify` trait. 

LineStrings are densified to ensure that a Line segment never has to be partitioned more than once. Additionally, after densification the lines are counted. If the number of lines is exactly `n`, then the function is exited early. 

An additional change was made which incorrectly would return the `Line` of a LineString if there were `n` lines. This was wrong because the length may have not been equivalent. I have removed this check and early return. I've removed a test that tested for this equivalence since it was a bad assumption.

I've also removed the unneeded `peekable()` call.

------

Aids downstream: 

- https://github.com/JosiahParry/rsgeo/issues/28
- https://github.com/ropensci/stplanr/issues/546

### CODE REVIEW

File: geo/src/algorithm/linestring_segment.rs [hunk 1]
```diff
@@ -78,6 +60,27 @@ impl LineStringSegmentize for LineString {
         let segment_prop = (1_f64) / (n as f64);
         let segment_length = total_length * segment_prop;
 
+        // densify the LineString so that each `Line` segment is not longer
+        // than the segment length ensuring that we will never partition one
+        // Line more than once.
+        let densified = self.densify(segment_length);
+
+        // if the densified lines is exactly equal to the number of requested
```
Comment (urschrei) [inline geo/src/algorithm/linestring_segment.rs:68]: Should be "line"

**Review by urschrei — APPROVED** [review]
Just a tiny typo and then I'm happy to merge.

File: geo/src/algorithm/linestring_segment.rs (hunk 1 above)
Comment (JosiahParry) [reply geo/src/algorithm/linestring_segment.rs]: et voila! Thank you @urschrei

**Comment by JosiahParry** [comment]
Thanks for your speedy help, @urschrei !!
